solutions/pytorch_tests/Dockerfile-build

```diff
@@ -0,0 +1,60 @@
+# Adapted from https://github.com/pytorch/pytorch/blob/master/Dockerfile
+ARG BASE_IMAGE=ubuntu:18.04
+
+FROM ${BASE_IMAGE} as dev-base
+RUN --mount=type=cache,id=apt-dev,target=/var/cache/apt \
+    apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        ccache \
+        cmake \
+        curl \
+        git \
+        libjpeg-dev \
+        libpng-dev && \
+    rm -rf /var/lib/apt/lists/*
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache && ccache --set-config=cache_dir=/opt/ccache
+ENV PATH /opt/conda/bin:$PATH
+
+FROM dev-base as conda
+ARG PYTHON_VERSION=3.8
+RUN curl -fsSL -v -o ~/miniconda.sh -O  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh  && \
+    chmod +x ~/miniconda.sh && \
+    ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    /opt/conda/bin/conda install -y python=${PYTHON_VERSION} conda-build pyyaml numpy ipython && \
+    /opt/conda/bin/pip install expecttest hypothesis typing_extensions pytest && \
+    /opt/conda/bin/conda clean -ya
+
+FROM dev-base as submodule-update
+ARG PYTORCH_VERSION=v1.10.0
+ENV PYTORCH_VERSION ${PYTORCH_VERSION}
+WORKDIR /opt
+RUN git clone -b v1.10.0 https://github.com/pytorch/pytorch.git --single-branch --depth 1
```

Comment:
I think you meant to refer to `PYTORCH_VERSION`. Though, what more does adding it as an environment variable achieve?

Reply:
Good catch! Fixed.